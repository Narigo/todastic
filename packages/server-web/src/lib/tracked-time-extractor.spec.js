const { extractTrackedTime } = require("./tracked-time-extractor");

describe("TrackedTimeExtractor", () => {
  it(
    "doesn't change anything when no tracked time is in the string",
    testExtraction("no time #really", [], "no time #really")
  );
  it(
    "can parse a simple example",
    testExtraction(
      'worked on #TRACK({"trackedTime": "01:00:00", "tracker": "Jörn", "date": "2018-09-13 17:27:00"})',
      [{ trackedTime: "01:00:00", tracker: "Jörn", date: "2018-09-13 17:27:00" }],
      "worked on"
    )
  );
});

function testExtraction(inputString, expectedEntries, expectedResultString) {
  return () => {
    const result = extractTrackedTime(inputString);
    expect(result.timeEntries).toEqual(expectedEntries);
    expect(result.text).toBe(expectedResultString);
  };
}
