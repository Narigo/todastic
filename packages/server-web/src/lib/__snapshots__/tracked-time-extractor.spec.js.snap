// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackedTimeExtractor can parse a simple example 1`] = `
Object {
  "text": "worked on",
  "trackedTimes": Array [
    Object {
      "date": "2018-09-13 17:27:00",
      "trackedTime": "01:00:00",
      "tracker": "Jörn",
    },
  ],
}
`;

exports[`TrackedTimeExtractor can parse more than one entry 1`] = `
Object {
  "text": "worked on #great",
  "trackedTimes": Array [
    Object {
      "date": "2018-09-15 18:20:15",
      "trackedTime": "00:10:00",
      "tracker": "Udo",
    },
    Object {
      "date": "2018-09-13 17:27:00",
      "trackedTime": "01:00:00",
      "tracker": "Jörn",
    },
  ],
}
`;

exports[`TrackedTimeExtractor discards broken entries 1`] = `
Object {
  "text": "worked on #great",
  "trackedTimes": undefined,
}
`;

exports[`TrackedTimeExtractor doesn't change anything when no tracked time is in the string 1`] = `
Object {
  "text": "no time #really",
  "trackedTimes": undefined,
}
`;
